.github/workflows/release.yml

```diff
@@ -30,3 +30,239 @@ jobs:
+
+      - name: Create release Pull Request
+        id: pr
+        env:
+          # NOTE: If we use the default GITHUB_TOKEN to create the release PR, the checks on the release PR will not be triggered automatically
+          GITHUB_TOKEN: ${{ secrets.RELEASE_GITHUB_TOKEN || github.token }}
+        uses: changesets/action@v1
+        with:
+          title: Version Packages (v2)
+          version: pnpm version-for-release
+
+      - name: Store hasChangesets for subsequent jobs
+        id: store-has-changesets
+        run: |
+          echo "hasChangesets=${{steps.pr.outputs.hasChangesets}}"
+          echo "hasChangesets=${{steps.pr.outputs.hasChangesets}}" >> $GITHUB_OUTPUT
+
+  pack:
+    name: Generate tarballs for publishing
+    needs: pr
+    if: needs.pr.outputs.hasChangesets == 'false'
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    outputs:
+      hasPackages: ${{ steps.detect.outputs.hasPackages }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v5
+
+      - name: Set up the environment
+        uses: ./.github/actions/setup-env
+
+      - name: Install Dependencies
+        run: pnpm install --frozen-lockfile
+
+      - name: Build All Packages
+        run: pnpm run --recursive --no-bail --filter './packages/**' --if-present build
+
+      - name: Publish All Packages (dry-run) and save list of packages that need publishing
+        run: pnpm publish --filter "./packages/**" -r --no-git-checks --access public --dry-run --report-summary
+
+      - name: Store the package list to be published
+        uses: actions/upload-artifact@v4
+        with:
+          name: pnpm-publish-summary.json
+          path: pnpm-publish-summary.json
+          overwrite: true
+          if-no-files-found: error
+
+      - name: Detect packages needing publish
+        id: detect
+        run: |
+          count=$(jq '.publishedPackages | length' pnpm-publish-summary.json 2>/dev/null)
+          if [ "$count" -gt 0 ]; then
+            echo "hasPackages=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "No packages to publish."
+            echo "hasPackages=false" >> "$GITHUB_OUTPUT"
+          fi
+
+      # The --filter parameter filters by exact package name, so there's no risk of a package name being a substring of another package name
+      - name: Generate tarballs for packages that need publishing
+        id: pack
+        if: steps.detect.outputs.hasPackages == 'true'
+        run: |
+          jq -r '.publishedPackages[].name' pnpm-publish-summary.json | while IFS= read -r name; do
+            pnpm pack -r --filter "$name" --pack-destination tarballs
+          done
+
+      - name: Upload packed tarballs
+        uses: actions/upload-artifact@v4
+        if: steps.detect.outputs.hasPackages == 'true'
+        with:
+          name: tarballs
+          path: tarballs
+          overwrite: true
+          if-no-files-found: error
+
+  review:
```

Comment:
I'm pretty sure this is incorrect, as it seems to be comparing with latest.

In the example repo i created the tag is a variable, so that would make it simpler to keep in sync.